com.unity.robotics.urdf-importer/Runtime/Extensions/UrdfRobotExtensions.cs

```diff
@@ -0,0 +1,363 @@
+            im.robot = new Robot(filename);
+
+            if (!UrdfAssetPathHandler.IsValidAssetPath(im.robot.filename))
+            {
+                Debug.LogError("URDF file and ressources must be placed in Assets Folder:\n" + Application.dataPath);
+                if (forceRuntimeMode) 
+                { // set runtime mode back to what it was
+                    RuntimeURDF.SetRuntimeMode(im.wasRuntimeMode);
+                }
+                return null;
+            }
+            return im;
+        }
+
+        // Creates the robot game object.
+        private static void ImportPipelineCreateObject(ImportPipelineData im) 
+        {
+            im.robotGameObject = new GameObject(im.robot.name);
+           
+            importsettings = im.settings;
+            im.settings.totalLinks = im.robot.links.Count;
+            CreateTag();
+            im.robotGameObject.tag = tagName;
+
+            im.robotGameObject.AddComponent<UrdfRobot>();
+
+            im.robotGameObject.AddComponent<RosSharp.Control.Controller>();
+            if (RuntimeURDF.IsRuntimeMode()) 
+            {// In runtime mode, we have to disable controller while robot is being constructed.
+                im.robotGameObject.GetComponent<RosSharp.Control.Controller>().enabled = false;
+            }
+
+            im.robotGameObject.GetComponent<UrdfRobot>().SetAxis(im.settings.choosenAxis);
+
+            UrdfAssetPathHandler.SetPackageRoot(Path.GetDirectoryName(im.robot.filename));
+            UrdfMaterial.InitializeRobotMaterials(im.robot);
+            UrdfPlugins.Create(im.robotGameObject.transform, im.robot.plugins);
+        }
+
+        // Creates the stack of robot joints. Should be called iteratively until false is returned.
+        private static bool ProcessJointStack(ImportPipelineData im)
+        {
+            if (im.importStack == null) 
+            {
+                im.importStack = new Stack<Tuple<Link, Transform, Joint>>();
+                im.importStack.Push(new Tuple<Link, Transform, Joint>(im.robot.root, im.robotGameObject.transform, null));
+            }
+            
+            if (im.importStack.Count != 0)
+            {
+                Tuple<Link, Transform, Joint> currentLink = im.importStack.Pop();
+                GameObject importedLink = UrdfLinkExtensions.Create(currentLink.Item2, currentLink.Item1, currentLink.Item3);
+                im.settings.linksLoaded++;
+                foreach (Joint childJoint in currentLink.Item1.joints)
+                {
+                    Link child = childJoint.ChildLink;
+                    im.importStack.Push(new Tuple<Link, Transform, Joint>(child, importedLink.transform, childJoint));
+                }
+                return true;
+            }
+            return false;
+        }
+
+        // Post creation stuff: add to parent, fix axis and add collision exceptions.
+        private static void ImportPipelinePostCreate(ImportPipelineData im)
+        {
+#if UNITY_EDITOR
```

Comment:
nit: a more general question that whether it would be better and more extensible to move split the code and move those in `UNITY_EDITOR` to the Editor directory?

Reply:
valid point. However that will require breaking the UrdfRobotExtensions class into two separate and changing the interface.